src/main/java/com/clivelee/crawler/EmailCrawler.java

```diff
@@ -0,0 +1,50 @@
+package com.clivelee.crawler;
+
+import com.clivelee.utils.TextFinder;
+import edu.uci.ics.crawler4j.crawler.Page;
+import edu.uci.ics.crawler4j.crawler.WebCrawler;
+import edu.uci.ics.crawler4j.parser.HtmlParseData;
+import edu.uci.ics.crawler4j.url.WebURL;
+
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+
+final class EmailCrawler extends WebCrawler {
+    private final static Pattern FILTERS = Pattern.compile("([.](css|js|html|htm|pdf))([?].*)?$");
```

Comment:
I was hoping for an explanation actually, but I was vague:  Why does the URL have to match this pattern?

It now seems to fix the case that I was looking for, which is URLs that end in / or otherwise don't have an extension. There's a lot of those on web.mit.edu that were being missed.  

I suspect your intention though was to factor in the optional query string portion of the URL, which was a good guess at what I was going for.  Can you explain what you were going for here, and/or take another crack at it?

Reply:
Ah, my original comment said "HTML documents", when I should have said something less ambiguous like: "HTML documents that end in .css, .html, .js, etc"